activerecord/lib/active_record/reflection.rb

```diff
@@ -708,10 +708,13 @@ def inverse_name
         def automatic_inverse_of
           if can_find_inverse_of_automatically?(self)
             inverse_name = ActiveSupport::Inflector.underscore(options[:as] || active_record.name.demodulize).to_sym
-            plural_inverse_name = ActiveSupport::Inflector.pluralize(inverse_name)
 
             begin
-              reflection = klass._reflect_on_association(inverse_name) || klass._reflect_on_association(plural_inverse_name)
+              reflection = klass._reflect_on_association(inverse_name)
```

Comment:
👍 for a deprecation warning, I think here is a good place to put it

Something like that?

```ruby
reflection_singular = klass._reflect_on_association(inverse_name)
reflection_plural = klass._reflect_on_association(plural_inverse_name)

if !reflection_singular && reflection_plural && klass.automatic_scope_inversing != :plurals
  puts "Deprecated"
end
```